Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamAVPictureInPictureVideoCallViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import AVKit
+import Foundation
+import StreamVideo
+import StreamWebRTC
+
+/// Describes an object that can be used to present picture-in-picture content.
+protocol StreamAVPictureInPictureViewControlling {
+    
+    /// The closure to call whenever the picture-in-picture window size changes.
+    var onSizeUpdate: ((CGSize) -> Void)? { get set }
+
+    /// The track that will be rendered on picture-in-picture window.
+    var track: RTCVideoTrack? { get set }
+
+    /// The preferred size for the picture-in-picture window.
+    /// - Important: This should **always** be greater to ``CGSize.zero``. If not, iOS throws
+    /// a cryptic error with content `PGPegasus code:-1003`
```

Comment:
that's what that pegasus code meant 😁

Reply:
Yeap and it was VERY difficult to figure it out. So I thought to add a note here so we can track it down (or anyone else searching the internet)

Reply:
ChatGPT will be able to answer it in 2 years 😄 cc sam